### PR TITLE
Use native geo_data instead of pycountry

### PR DESCRIPTION
**Pull request by viranch**
Dropping pycountry rids of many packages, speeding up builds and pip installs of octodns-ns1.

### CODE REVIEW

File: octodns_ns1/__init__.py [hunk 1]
```diff
@@ -690,17 +699,7 @@ def _parse_rule_geos(self, meta, notes):
...
-            # Sint Maarten ('SX')
-            if country == 'TL':
-                con = 'AS'
-            elif country == 'SX':
-                con = 'NA'
-            elif country in ('PN', 'UM'):
-                con = 'OC'
-            else:
-                con = country_alpha2_to_continent_code(country)
+            con = country_alpha2_to_continent_code(country)
```
Comment (ross) [inline octodns_ns1/__init__.py:702]: Just double checked that these all map out the same in the new reversed version of things.

File: octodns_ns1/__init__.py [hunk 2]
```diff
@@ -25,6 +24,16 @@ def _ensure_endswith_dot(string):
     return string if string.endswith('.') else f'{string}.'
 
 
+country_to_continent = {}
+for continent, countries in geo_data.items():
+    for country in countries.keys():
+        country_to_continent[country] = continent
```
Comment (ross) [inline octodns_ns1/__init__.py:30]: Make sense to build this on demand, first time it's requested, or lightweight enough not to worry about it?

**Review by ross — APPROVED** [review]

File: octodns_ns1/__init__.py (hunk 2 above)
Comment (viranch) [reply octodns_ns1/__init__.py:30]: Thinking more about it, octodns' `geo.py` already has utility methods that can be re-used here. Let me put that commit in.

File: octodns_ns1/__init__.py (hunk 1 above)
Comment (viranch) [reply octodns_ns1/__init__.py:702]: Yeah the test for those special test cases is also passing.

File: octodns_ns1/__init__.py [hunk 3]
```diff
@@ -694,22 +693,13 @@ def _parse_rule_geos(self, meta, notes):
...
-            if country == 'TL':
-                con = 'AS'
-            elif country == 'SX':
-                con = 'NA'
-            elif country in ('PN', 'UM'):
-                con = 'OC'
-            else:
-                con = country_alpha2_to_continent_code(country)
+            geo_code = GeoCodes.country_to_code(country)
+            con = GeoCodes.parse(geo_code)['continent_code']
```
Comment (viranch) [inline octodns_ns1/__init__.py:697]: This implementation is on-demand, but its not all that computationally expensive so should be ok.

Comment (ross) [reply octodns_ns1/__init__.py:697]: I've thrown a personal TODO in to create a pre/purpose-built version of this in geo_data just b/c there's no reason not to.

**Review by ross — APPROVED** [review]

File: octodns_ns1/__init__.py (hunk 3 above)
Comment (viranch) [reply octodns_ns1/__init__.py:697]: 👍 fwiw the 2-step thing here is needed since `geo_code` is used below.

**Pull request merged**